pulsar-broker-common/src/main/java/org/apache/pulsar/broker/web/AuthenticationFilter.java

```diff
@@ -52,19 +52,29 @@ public AuthenticationFilter(AuthenticationService authenticationService) {
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
             throws IOException, ServletException {
+        boolean allowed = false;
+        Exception authenticationException = null;
         try {
-            boolean doFilter = authenticationService
+            allowed = authenticationService
```

Comment:
Nit: `allowed` is a misleading variable name here. From my perspective, the correct name was `doFilter`. In some cases, like in SASL, it returns `false` because there is additional work to be done before proceeding. I don't think anything is incorrect about the logic, just commenting on the name.

See:

https://github.com/apache/pulsar/blob/82237d3684fe506bcb6426b3b23f413422e6e4fb/pulsar-broker-auth-sasl/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderSasl.java#L229-L312

Reply:
Then let me take my comment above and a follow-up https://github.com/apache/pulsar/pull/20598.

Reply:
You can take a look at the PR. You can actually `return` from the catch block.

Reply:
Oops. I notice that it's a boolean return value..Let me update it.

Reply:
Updated. Let's discuss in #20598 .